codalab/lib/bundle_cli.py

```diff
@@ -2516,7 +2516,7 @@ def follow_targets(self, client, bundle_uuid, subpaths, from_start=False):
                     if not result:
                         break
                     subpath_offset[i] += len(result)
-                    self.stdout.write(result)
+                    self.stdout.write(ensure_str(result))
```

Comment:
Is `result` only sometimes a string and sometimes binary?

if that's not the case, then we shouldn't use `ensure_str` and we should instead do the necessary encoding / decoding directly.

Reply:
In this case, I can only reproduce `bytes` during the experiments that I was running (not sure if there will be other cases that will happen in the future). 
I agree that we should do necessary encoding or decoding directly. 
The only thing that makes me decide to use `ensure_str` function is that I learned it from a past task that there could be `UnicodeDecodeError` happening during the `decode` phase, and `ensure_str` contains the error handling when `UnicodeDecodeError` happens. I think we can reuse this piece of code even though it has one additional checking on if the current `result` is a string, instead of rewriting duplicate code to handle the exception. 
If we think we want to handle decoding directly, I am open to that.